src/IronPythonExtension/IronPythonExtension.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.IO;
+using System.Linq;
+using System.Reflection;
+using Dynamo.Extensions;
+
+namespace IronPythonExtension
+{
+    public class IronPythonExtension : IExtension
+    {
+        private const string PythonEvaluatorAssembly = "DSIronPython";
+
+        public void Dispose()
+        {
+        }
+
+        public string UniqueId => "D7B449D7-4D54-47EF-B742-30C7BEDFBE92";
+
+        public string Name => "IronPythonExtension";
+
+        public void Startup(StartupParams sp)
+        {
+            // Searches for python node libraries in package paths
+            var packagePaths = sp.Preferences.CustomPackageFolders;
+            string pythonFile = string.Empty;
+            foreach (var path in packagePaths)
+            {
+                //var pkgDir = Path.GetDirectoryName(path);
+                pythonFile = Directory.EnumerateFiles(path, "" + ".dll",
```

Comment:
Since we have control over this package why don't we simply save the dll in the bin directory of the package and then point `pythonDir` to that instead of searching for the directory that contains any dlls?

Reply:
Coming to think of it we are writing an extension to function as a package :)
It looks like we can just make this a ZT package and therefore avoid having to write all this code 🤔

Reply:
Noted in my TODO, not ready for review yet

Reply:
We've gone over the package vs extension issues a few times.

One consideration is that currently extensions can easily be loaded OOTB just by putting the extension manifest into the extensions folder.

Dynamo cannot current load packages by default, we have no location for them except for user's package folders.